choicepy/__init__.py

```diff
@@ -1,593 +1,616 @@
+        else:
+            raise StopIteration
+
+    def __eq__(self,other):
+        # equality check
+        if isinstance(other, list):
+            return self.voters == other
+
+        if isinstance(other, Profile):
+            return self.voters == other.voters
+    def __str__(self):
+        term_rows, term_columns = list(map(int,(os.popen('stty size', 'r').read().split())))
+
+        max_length = len(max(self.voters[0], key=len))
+        justify_length = max_length + 3
+
+        voters_printable = str(self.num_candidates) + " candidates, " + str(self.num_voters) + " voters \n"
+        voters_printable +="Profile: " + str(self.print_summary()) + "\n"
+        if self.num_voters * justify_length <= term_columns:
+            voters_printable += + "\n"
+
+
+            voters_printable += "".join([("[" + str(item) + "]").ljust(justify_length) for item in range(self.num_voters)])
+            voters_printable += "\n"
+            for i in range(self.num_candidates):
+                for j in range(self.num_voters):
+
+                    voters_printable += self.voters[j][i].ljust(justify_length)
+                voters_printable += "\n"
+
+
+        return voters_printable
+        
+
+    def __repr__(self):
+       return self.__str__()
+
+    def print_full(self):
+        term_rows, term_columns = list(map(int,(os.popen('stty size', 'r').read().split())))
+
+        max_length = len(max(self.voters[0], key=len))
+        justify_length = max_length + 3
+
+        voters_printable = str(self.num_candidates) + " candidates, " + str(self.num_voters) + " voters \n"
+        voters_printable +="Profile: " + str(self.print_summary()) + "\n\n"        
+        if True:
+
+            voters_printable += "".join([("[" + str(item) + "]").ljust(justify_length) for item in range(self.num_voters)])
+            voters_printable += "\n"
+            for i in range(self.num_candidates):
+                for j in range(self.num_voters):
+
+                    voters_printable += self.voters[j][i].ljust(justify_length)
+                voters_printable += "\n"
+
+
+        print(voters_printable)
+
+
+
+    def set_voters(self, voter_list):
+        self.voters = voter_list
+        self.num_voters = len(self.voters)
+        self.set_candidates(sorted(self.voters[0]))
+
+    def set_candidates(self, candidates):
+        """
+        Sets candidate list.
+        Parameters
+        ----------
+        candidates: Either a list or an int.
+            If candidates is a list, it just sets self.candidates as this list;
+            if candidates is an int, it will create a list of length candidates with lexicographically ordered strings
+            ("a", "b", "c", ..., "z", "aa", "ab", "ac", ...)
+        """
+        try:
+            self.candidates = sorted(candidates)
+        except TypeError:
+            repetitions = range(candidates // 26 + 2)
+            candidate_tuples = itertools.chain.from_iterable(itertools.product(
```

Comment:
maybe we can move this generation to the function `get_alphabet_first_n` function I commented on above. moreover I am not sure whether we should use try and except instead of ifs. 

I would do something like:
```
if isinstance(candidates,list):
    self.candiates = sorted(candidates)
elsif isinstance(candidates, int):                    # or double?
   self.candidates = get_lexicographic_list(candidates)       # I've just made the name up

Reply:
Yes, you're right! I just read that you only should use try-except blocks if exceptions really are exceptional, which they are rather not in our case. I will change that and move it to the other function.

And I am also not sure whether to check if the input is an instance of double instead of int. I think it would be nice if also a number like 5.0 would be recognized as an int. And I guess that also inputs like 5.3 should not cause any problems.